### PR TITLE
feat!: Do not treat Buffers as JSON by default

### PR DESCRIPTION
**Pull request by danielbankhead**
Dependency:
- #617

🦕

### CODE REVIEW

**Comment by sofisl** [comment]
Is there a FR tied to this?

**Comment by danielbankhead** [comment]
@sofisl more-so bug reports and difficult to catch behavior:
- https://github.com/googleapis/gaxios/issues/348
- https://github.com/googleapis/gaxios/issues/447
- https://github.com/googleapis/gaxios/issues/156

File: test/test.getch.ts [hunk 1]
```diff
@@ -1112,22 +1112,6 @@ describe('🍂 defaults & instances', () => {
     assert.deepStrictEqual(res.data, {});
   });
 
-  it('should set content-type to application/json by default, for buffer', async () => {
```
Comment (sofisl) [inline test/test.getch.ts:1115]: can we assert a test for what happens if a buffer does get sent in?

**Review by sofisl — CHANGES_REQUESTED** [review]

File: test/test.getch.ts (hunk 1 above)
Comment (danielbankhead) [reply test/test.getch.ts:1115]: Sure, it’s mostly a partial revert of:
- https://github.com/googleapis/gaxios/pull/374/files

but that can be added.

**Review by sofisl — APPROVED** [review]

**Pull request merged**